compiler/pathutils.nim

```diff
@@ -26,6 +26,14 @@ proc copyFile*(source, dest: AbsoluteFile) =
 
 proc removeFile*(x: AbsoluteFile) {.borrow.}
 
+template toRelative(x: typedesc[AnyPath]): typedesc =
+  when x is AbsoluteFile | RelativeFile: RelativeFile else: RelativeDir
+
+proc lastPathPart*(x: AnyPath): auto =
```

Comment:
The following rule served me well: If you cannot write down the return type what you wrote is not a ``proc`` or ``func``, but something else.

Reply:
I've removed `lastPathPart`